rusoto/core/src/signature.rs

```diff
@@ -260,9 +271,9 @@ impl SignedRequest {
         self.remove_header("Content-Type");
 
         if let Some(ref token) = *creds.token() {
-            self.remove_header("x-amz-security-token");
+            self.remove_header("X-Amz-Security-Token");
```

Comment:
The case matters. Otherwise, STS tokens will encounter "invalid security token" errors.